lib/Service/LiveTranscriptionService.php

```diff
@@ -0,0 +1,254 @@
+			if ($e->getResponse()->getStatusCode() !== 404) {
+				throw $e;
+			}
+		}
+
+		$this->roomService->setLiveTranscriptionLanguageId($room, $languageId);
+	}
+
+	/**
+	 * @throws LiveTranscriptionAppNotEnabledException if the external app
+	 *                                                 "live_transcription" is
+	 *                                                 not enabled.
+	 * @throws LiveTranscriptionAppAPIException if the request could not be sent
+	 *                                          to the app or the response could
+	 *                                          not be processed.
+	 * @throws LiveTranscriptionAppResponseException if the request itself
+	 *                                               succeeded but the app
+	 *                                               responded with an error.
+	 */
+	private function requestToExAppLiveTranscription(string $method, string $route, array $parameters = []): ?array {
+		try {
+			$appApiPublicFunctions = $this->getAppApiPublicFunctions();
+		} catch (LiveTranscriptionAppAPIException $e) {
+			if ($e->getMessage() === 'app-api') {
+				$this->logger->error('AppAPI is not enabled');
+			} elseif ($e->getMessage() === 'app-api-functions') {
+				$this->logger->error('Could not get AppAPI public functions', ['exception' => $e]);
+			}
+
+			throw new LiveTranscriptionAppNotEnabledException($e->getMessage());
+		}
+
+		if (!$this->isLiveTranscriptionAppEnabled($appApiPublicFunctions)) {
+			$this->logger->error('live_transcription (ExApp) is not enabled');
+
+			throw new LiveTranscriptionAppNotEnabledException('live-transcription-app');
+		}
+
+		$response = $appApiPublicFunctions->exAppRequest(
+			'live_transcription',
+			$route,
+			$this->userId,
+			$method,
+			$parameters,
+		);
+
+		if (is_array($response) && isset($response['error'])) {
+			$this->logger->error('Request to live_transcription (ExApp) failed: ' . $response['error']);
+
+			throw new LiveTranscriptionAppAPIException('response-error');
+		}
+
+		if (is_array($response)) {
+			// AppApi only uses array responses for errors, so this should never
+			// happen.
+			$this->logger->error('Request to live_transcription (ExApp) failed: response is not a valid response object');
+
+			throw new LiveTranscriptionAppAPIException('response-invalid-object');
+		}
+
+		$responseContentType = $response->getHeader('Content-Type');
+		if (strpos($responseContentType, 'application/json') !== false) {
+			$body = $response->getBody();
+			if (!is_string($body)) {
+				$this->logger->error('Request to live_transcription (ExApp) failed: response body is not a string, but content type is application/json', ['response' => $response]);
+
+				throw new LiveTranscriptionAppAPIException('response-content-type');
+			}
+
+			$decodedBody = json_decode($body, true);
+		} else {
+			$decodedBody = ['response' => $response->getBody()];
+		}
+
+		if (intval($response->getStatusCode() / 100) !== 2) {
```

Comment:
Please don't do math with HTTP status codes. We should know valid values and if not check `< 200 || >= 300` (but redirect could also be fine?)

Reply:
Sorry, copy/paste leftover from https://github.com/nextcloud/context_chat/blob/2ea9768bec56d0ea3dbe1551d3680b77f6ea48f4/lib/Service/LangRopeService.php#L152 :-)

I am not sure about redirects. Wouldn't it need an additional request to fetch the actual contents in that case? In any case I set it to `< 200 || >= 300` as currently the `live_transcription` app [does not seem to use 3XX responses](https://github.com/nextcloud/live_transcription/blob/db3a03dfa3d3fec4cfcd2429ffdf897d334e6ec8/ex_app/lib/main.py).